contracts/base-chain/bridge-gateways/AMPLArbitrumGateway.sol

```diff
@@ -0,0 +1,388 @@
+        uint256 recordedGlobalAMPLSupply = IERC20(ampl).totalSupply();
+
+        emit XCRebaseReportOut(recordedGlobalAmpleforthEpoch, recordedGlobalAMPLSupply);
+
+        uint256 seqNumber = sendTxToL2(
+            inbox,
+            counterpartGateway,
+            tx.origin,
+            msg.value,
+            0,
+            L2GasParams({
+                _maxSubmissionCost: _maxSubmissionCost,
+                _maxGas: _maxGas,
+                _gasPriceBid: _gasPriceBid
+            }),
+            abi.encodeWithSelector(
+                ArbitrumXCAmpleGateway.reportRebaseCommit.selector,
+                recordedGlobalAmpleforthEpoch,
+                recordedGlobalAMPLSupply
+            )
+        );
+
+        emit RebaseReportInitiated(seqNumber);
+
+        return abi.encode(seqNumber);
+    }
+
+    /**
+     * @notice Deposit AMPL from Ethereum into Arbitrum.
+     * @param _l1Token L1 address of the AMPL ERC20.
+     * @param _to account to be credited with the tokens in the L2 (can be the user's L2 account or a contract).
+     * @param _amount Token Amount.
+     * @param _maxGas Max gas deducted from user's L2 balance to cover L2 execution.
+     * @param _gasPriceBid Gas price for L2 execution.
+     * @param _data encoded data from router and user.
+     * @return res abi encoded inbox sequence number.
+     */
+    function outboundTransfer(
+        address _l1Token,
+        address _to,
+        uint256 _amount,
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        bytes calldata _data
+    ) external payable override returns (bytes memory) {
+        require(msg.sender == router, "AMPLArbitrumGateway: NOT_FROM_ROUTER");
+
+        require(_l1Token == ampl, "AMPLArbitrumGateway: ONLY_AMPL_ALLOWED");
+
+        require(xcAmple != address(0), "AMPLArbitrumGateway: NOT_INITIALIZED");
+
+        L2GasParams memory gasParams = L2GasParams({
+            _maxSubmissionCost: 0,
+            _maxGas: _maxGas,
+            _gasPriceBid: _gasPriceBid
+        });
+
+        address from;
+        (from, gasParams._maxSubmissionCost) = _parseDataFromRouterOnTransfer(_data);
+
+        // Lock funds and log outbound transfer
+        uint256 recordedGlobalAMPLSupply;
+        {
+            recordedGlobalAMPLSupply = IERC20(_l1Token).totalSupply();
+
+            // NOTE: The usual xc-transfer flow involves the depositer approving the vault
+            //       and initiating the transfer. However the arbitrum implementation expects
+            //       the user to approve the gateway. We thus add this extra step to confirm
+            //       to both interfaces.
+            //       1) User approves the gateway
+            //       2) Tokens transfer from user => gateway => vault
+            IERC20(_l1Token).safeTransferFrom(from, address(this), _amount);
+            IERC20(_l1Token).approve(vault, _amount);
+
+            ITokenVault(vault).lock(_l1Token, address(this), _amount);
```

Comment:
We do this to keep the audit trail on the vault. Explicitly calling `lock` over just transferring directly into the vault logs the `GatewayLocked` event.

https://github.com/ampleforth/cross-chain-ample/blob/master/contracts/base-chain/TokenVault.sol#L31

However I wonder if the `GatewayLocked` and `GatewayUnlocked` events are redundant as one could always reconstruct the same information using merely the token transfer logs. 

If so we can `safeTransferFrom` the user's wallet directly into the vault which seems cleaner.